### PR TITLE
makes it possible to have implicit QueryStringBindable's for AnyVal

### PR DESCRIPTION
**Pull request by schmitch**
added a way to do the thing I commited for `PathBindable`'s to `QueryStringBindable`'s.

### CODE REVIEW

File: documentation/manual/working/scalaGuide/main/http/ScalaRouting.md [hunk 1]
```diff
@@ -118,6 +118,19 @@ And do the same on the corresponding `show` method definition in the `controller
...
+Play supports the following Parameter Types:
+
+- String
+- Int
+- Long
+- Double
+- Float
+- Boolean
+- UUID
+- AnyVal Types
```
Comment (PromanSEW) [inline documentation/manual/working/scalaGuide/main/http/ScalaRouting.md]: Where is Java `Optional`? See #8056

Comment (gmethvin) [inline documentation/manual/working/scalaGuide/main/http/ScalaRouting.md]: I would say something like "AnyVal wrappers for other supported types".

File: documentation/manual/working/scalaGuide/main/http/ScalaRouting.md [hunk 2]
```diff
@@ -118,6 +118,19 @@ And do the same on the corresponding `show` method definition in the `controller
...
+
+If you have a different type and want to implement it you can take a look at [[Request Binders|ScalarequestBinders]]
```
Comment (gmethvin) [inline documentation/manual/working/scalaGuide/main/http/ScalaRouting.md]: The link should be `ScalaRequestBinders`.

**Comment by schmitch** [comment]
applied suggestions

**Comment by francisdb** [comment]
We have a nonEmptyStringBindable that results in a `case class NonEmptyString(value: String) extends AnyVal` and with this change it no longer worked as implicitly requesting `QueryStringBindable[NonEmptyString]` would result in the play anyValQueryStringBindable. (workaround was to remove the extends AnyVal)

Maybe something to mention in the migration guide

**Pull request merged**